### PR TITLE
Feature/track users selection

### PR DESCRIPTION
**Pull request by m0e33**
- registers the selection in local browser and sends broadcast to all collaborators in a session

### CODE REVIEW

File: packages/CollabCodeEdit-Client.package/CCEBrowser.class/instance/editSelection..st [hunk 1]
```diff
@@ -0,0 +1,8 @@
+overwrite
+editSelection: aSelection
+	"Set the editSelection as requested."
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEBrowser.class/instance/editSelection..st:3]: Do we need that comment?

File: packages/CollabCodeEdit-Client.package/CCEBrowser.class/instance/editSelection..st [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+	editSelection := aSelection.
+	self changed: #editSelection.
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEBrowser.class/instance/editSelection..st]: Please delete the empty line here. Maybe use a cascade.

File: packages/CollabCodeEdit-Client.package/CCECollaborator.class/instance/currentSelectionAsDictionary..st [hunk 3]
```diff
@@ -0,0 +1,3 @@
+accessing
+currentSelectionAsDictionary: anObject
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCECollaborator.class/instance/currentSelectionAsDictionary..st]: Please specify.

File: packages/CollabCodeEdit-Client.package/CCECollaborator.class/instance/printOn..st [hunk 4]
```diff
@@ -1,6 +1,12 @@
...
\ No newline at end of file
+		nextPutAll: self identifier;
+		nextPutAll: ' (';
+		nextPutAll: self consistencyState;
+		nextPutAll: ') ';
+		nextPutAll: (self currentSelectionAsDictionary at: #categorySelection) asString;
+		nextPutAll: '>>';
+		nextPutAll: (self currentSelectionAsDictionary at: #classSelection) asString;
+		nextPutAll: '>>';
+		nextPutAll: (self currentSelectionAsDictionary at: #methodSelection) asString.
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCECollaborator.class/instance/printOn..st]: Wouldn't it be possible to concat all those string with `,`?

File: packages/CollabCodeEdit-Client.package/CCEModel.class/instance/CCEchangeSelection..st [hunk 5]
```diff
@@ -0,0 +1,3 @@
+requests backend
+CCEchangeSelection: anOrderedDictionary
+	self session changeSelectionWith: anOrderedDictionary.
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEModel.class/instance/CCEchangeSelection..st]:
```suggestion
	self session changeSelectionWith: anOrderedDictionary
```

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/^equals.st [hunk 6]
```diff
@@ -0,0 +1,5 @@
+comparing
+= aCCESelection
+	^ ( self categorySelection = aCCESelection categorySelection)  &
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/^equals.st]: Please use `and:` and blocks.

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/categorySelection..st [hunk 7]
```diff
@@ -0,0 +1,4 @@
+accessing
+categorySelection: anObject
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/categorySelection..st]: Should be specified.

Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/categorySelection..st]: Please delete the empty line.

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/categorySelection.st [hunk 8]
```diff
@@ -0,0 +1,4 @@
+accessing
+categorySelection
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/categorySelection.st]:
```suggestion
```

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/classSelection..st [hunk 9]
```diff
@@ -0,0 +1,4 @@
+accessing
+classSelection: anObject
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/classSelection..st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/classSelection..st [hunk 10]
```diff
@@ -0,0 +1,4 @@
...
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/classSelection..st]:
```suggestion
```

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/classSelection.st [hunk 11]
```diff
@@ -0,0 +1,4 @@
+accessing
+classSelection
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/classSelection.st]:
```suggestion
```

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/client..st [hunk 12]
```diff
@@ -0,0 +1,3 @@
+accessing
+client: anObject
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/client..st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/imageAccount..st [hunk 13]
```diff
@@ -0,0 +1,3 @@
+accessing
+imageAccount: anObject
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/imageAccount..st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/methodSelection..st [hunk 14]
```diff
@@ -0,0 +1,4 @@
+accessing
+methodSelection: anObject
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/methodSelection..st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/methodSelection..st [hunk 15]
```diff
@@ -0,0 +1,4 @@
...
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/methodSelection..st]:
```suggestion
```

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/methodSelection.st [hunk 16]
```diff
@@ -0,0 +1,4 @@
+accessing
+methodSelection
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/methodSelection.st]:
```suggestion
```

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/printOn..st [hunk 17]
```diff
@@ -0,0 +1,5 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/printOn..st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/printOn..st [hunk 18]
```diff
@@ -0,0 +1,5 @@
...
+printOn: aStream
+	aStream nextPutAll: self classSelection asString;
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/printOn..st]: Again, that should be possible with String concats.

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/updateSelectionWith..st [hunk 19]
```diff
@@ -0,0 +1,9 @@
+accessing
+updateSelectionWith: aBrowser
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/updateSelectionWith..st:3]:
```suggestion
```

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/updateSelectionWith..st [hunk 20]
```diff
@@ -0,0 +1,9 @@
...
+	self categorySelection: aBrowser selectedSystemCategoryName;
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/updateSelectionWith..st]:
```suggestion
	self 
	  categorySelection: aBrowser selectedSystemCategoryName;
```

File: packages/CollabCodeEdit-Client.package/CCESelection.class/instance/updateSelectionWith..st [hunk 21]
```diff
@@ -0,0 +1,9 @@
...
+		classSelection: aBrowser selectedClassName;
+		methodSelection: aBrowser selectedMessageName.
+		
+	Transcript show: self methodSelection.
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelection.class/instance/updateSelectionWith..st:9]: Debug code should not be in develop.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/browserModels.st [hunk 22]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/browserModels.st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/currentDefaultBrowserClass.st [hunk 23]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/currentDefaultBrowserClass.st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/currentSelection..st [hunk 24]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/currentSelection..st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/currentSelection..st [hunk 25]
```diff
@@ -0,0 +1,4 @@
...
+currentSelection: aCCESelection
+	currentSelection := aCCESelection
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/currentSelection..st]: Please delete the empty lines.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/currentSelection.st [hunk 26]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/currentSelection.st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/initialize.st [hunk 27]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+initialize
+	super initialize
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/initialize.st]: I don't see the point of this method.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/initialize.st [hunk 28]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/initialize.st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/start.st [hunk 29]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/start.st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/step.st [hunk 30]
```diff
@@ -0,0 +1,5 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/step.st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/step.st [hunk 31]
```diff
@@ -0,0 +1,5 @@
...
+step
+	self currentDefaultBrowserClass allInstances do: [:browser | 
+		browser addDependent: self].
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/step.st]: Please delete the empty line.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/stepTime.st [hunk 32]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/stepTime.st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/stop.st [hunk 33]
```diff
@@ -0,0 +1,5 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/stop.st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/update..st [hunk 34]
```diff
@@ -0,0 +1,5 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/update..st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/update..st [hunk 35]
```diff
@@ -0,0 +1,5 @@
...
+update: anObject 
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/update..st]:
```suggestion
```

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/update..st [hunk 36]
```diff
@@ -0,0 +1,5 @@
+as yet unclassified
+update: anObject
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/update..st:2]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/update..st [hunk 37]
```diff
@@ -0,0 +1,5 @@
+as yet unclassified
+update: anObject 
+	
+	(anObject isKindOf: Browser) ifFalse: [^ self].
+	self currentSelection updateSelectionWith: anObject.
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESelectionListener.class/instance/update..st]:
```suggestion
	self currentSelection updateSelectionWith: anObject
```

File: packages/CollabCodeEdit-Client.package/CCESession.class/instance/ackCreateSession..st [hunk 38]
```diff
@@ -1,6 +1,7 @@
-services
+session-managment
 ackCreateSession: anOrderedDictionary
 	self token: (anOrderedDictionary at: #token).
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESession.class/instance/ackCreateSession..st]: You should be able to use cascades here.

File: packages/CollabCodeEdit-Client.package/CCESession.class/instance/ackJoinSession..st [hunk 39]
```diff
@@ -18,5 +18,6 @@ ackJoinSession: anOrderedDictionary
 		ifFalse: [			
 			self setDirtySessionWith: remoteSessionScope and: remoteSessionHash.
 			self setConsistencyOfCollaborator: (self imageAccount humanReadableName) to: false.
-			self client sendMsg: (CCEMessage setConsistent: false) ]
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESession.class/instance/ackJoinSession..st]: `sendMsg` should be `sendMessage`.

Comment (arminWel) [reply packages/CollabCodeEdit-Client.package/CCESession.class/instance/ackJoinSession..st]: And while your at it: This method is quite long.

File: packages/CollabCodeEdit-Client.package/CCESession.class/instance/activateSelectionListener.st [hunk 40]
```diff
@@ -0,0 +1,10 @@
+selection managment
+activateSelectionListener
+	| cceSelection |
+	
+	cceSelection := CCESelection new
+		imageAccount: self imageAccount;
+		client: self client.
+
+	self selectionListener currentSelection: cceSelection;
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESession.class/instance/activateSelectionListener.st:9]:
```suggestion
	self selectionListener 
	  currentSelection: cceSelection;
```

File: packages/CollabCodeEdit-Client.package/CCESession.class/instance/changeSelectionWith..st [hunk 41]
```diff
@@ -0,0 +1,8 @@
+selection managment
+changeSelectionWith: anOrderedDictionary
+	| identifier selection | 
+	
+	identifier := ( anOrderedDictionary at: #id ).
+	selection := ( anOrderedDictionary at: #selection).
+	
+	self setSelectionOfCollaborator: identifier to: selection
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESession.class/instance/changeSelectionWith..st:8]: I guess that in #22 as well, but do you need those tmp vars?

File: packages/CollabCodeEdit-Client.package/CCESession.class/instance/selectionListener..st [hunk 42]
```diff
@@ -0,0 +1,3 @@
+accessing
+selectionListener: anObject
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESession.class/instance/selectionListener..st]: Should be specified.

File: packages/CollabCodeEdit-Client.package/CCESession.class/instance/setSelectionOfCollaborator.to..st [hunk 43]
```diff
@@ -0,0 +1,4 @@
+collaborator managment
+setSelectionOfCollaborator: aString to: anOrderedDictionary
+	( self collaborators at: aString ) currentSelectionAsDictionary: anOrderedDictionary.
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESession.class/instance/setSelectionOfCollaborator.to..st]:
```suggestion
	(self collaborators at: aString) currentSelectionAsDictionary: anOrderedDictionary.	
```

File: packages/CollabCodeEdit-Core.package/CCEMessage.class/class/changeSelectionBroadcastWith..st [hunk 44]
```diff
@@ -0,0 +1,4 @@
+server-Message creation
+changeSelectionBroadcastWith: anOrderedDictionary
+	^ self new: (CCEMessage serverMsgTypes at: #changeSelection:)
```
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEMessage.class/class/changeSelectionBroadcastWith..st]: Should be `serverMessageTypes` in develop.

File: packages/CollabCodeEdit-Core.package/CCEMessage.class/class/changeSelectionWith..st [hunk 45]
```diff
@@ -0,0 +1,4 @@
+client-Message creation
+changeSelectionWith: anOrderedDictionary
+	^ self new: (CCEMessage clientMsgTypes at: #changeSelection:from:)
```
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEMessage.class/class/changeSelectionWith..st:3]: `clientMessageTypes` would be more squeaky.

File: packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/instance/category..st [hunk 46]
```diff
@@ -0,0 +1,3 @@
+acc	essing
```
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/instance/category..st]: Is that an error of git?

File: packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/instance/category.st [hunk 47]
```diff
@@ -0,0 +1,3 @@
+acc	essing
```
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/instance/category.st]: See above.

File: packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/instance/method..st [hunk 48]
```diff
@@ -0,0 +1,3 @@
+acc	essing
```
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/instance/method..st]: S.a.

File: packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/instance/method.st [hunk 49]
```diff
@@ -0,0 +1,3 @@
+acc	essing
```
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/instance/method.st]: S.a.

File: packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/properties.json [hunk 50]
```diff
@@ -0,0 +1,14 @@
...
+		 ],
+	"classvars" : [
+		 ],
+	"commentStamp" : "AD 6/24/2019 11:44",
+	"instvars" : [
+		 ],
+	"name" : "CCEMethodRecategorizedEdit",
+	"pools" : [
+		 ],
+	"super" : "AnObsoleteCCEEdit",
```
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/properties.json]: This should be `CCEEdit`. Please check, that the `CCEEdit` is not deleted in your branch.

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEchangeSelection.from..st [hunk 51]
```diff
@@ -0,0 +1,4 @@
+request handling
+CCEchangeSelection: anOrderedDictionary from: aClient
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEchangeSelection.from..st]:
```suggestion
```

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEchangeSelection.from..st [hunk 52]
```diff
@@ -0,0 +1,4 @@
+request handling
+CCEchangeSelection: anOrderedDictionary from: aClient
+	
+	aClient session broadcastMsg: (CCEMessage changeSelectionBroadcastWith: anOrderedDictionary)
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEchangeSelection.from..st:3]: `broadcastMsg:` is `broadcastMessage` in develop.

File: packages/CollabCodeEdit-Tests.package/CCEProtocolCollaboratorsSSTest.class/instance/subClassSpecificTearDown.st [hunk 53]
```diff
@@ -0,0 +1,4 @@
+running
+subClassSpecificTearDown
+	
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolCollaboratorsSSTest.class/instance/subClassSpecificTearDown.st:4]: Remove the trailing lines please.

File: packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/subClassSpecificSetUp.st [hunk 54]
```diff
@@ -0,0 +1,17 @@
...
+	remoteId := 'testId'.
+	
+	changeSelectionPayload := OrderedDictionary newFrom: {
+		#id -> remoteId  . 
+		#selection -> (OrderedDictionary newFrom: {
+			#categorySelection -> remoteSelection categorySelection .
+			#classSelection -> remoteSelection classSelection .
+			#methodSelection -> remoteSelection methodSelection }) }.
+	
+	session collaborators at: remoteId put: ( CCECollaborator new identifier: remoteId; consistencyState: CCECollaborator isConsistentString ).
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/subClassSpecificSetUp.st]: Please remove the trailing line.

File: packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/subClassSpecificSetUp.st [hunk 55]
```diff
@@ -0,0 +1,17 @@
+running
+subClassSpecificSetUp
+	cceSelection := CCESelection new client: client; imageAccount: imageAccount.
+	
+	browser := Browser new setClass: CCESelection selector: #categorySelection.
+	
+	remoteSelection := CCESelection new categorySelection: #testCategory; classSelection: #testClass; methodSelection: #testMethod.
+	remoteId := 'testId'.
+	
+	changeSelectionPayload := OrderedDictionary newFrom: {
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/subClassSpecificSetUp.st]: Maybe put that in an extra method.

File: packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/subClassSpecificTearDown.st [hunk 56]
```diff
@@ -0,0 +1,3 @@
+running
+subClassSpecificTearDown
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/subClassSpecificTearDown.st:3]:
```suggestion
```

File: packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/testChangeSelectionUpdatesSelectionInCollaboratorsList.st [hunk 57]
```diff
@@ -0,0 +1,12 @@
+tests
+testChangeSelectionUpdatesSelectionInCollaboratorsList
+	| selectionMessage | 
+	
+	selectionMessage := CCEMessage changeSelectionBroadcastWith: changeSelectionPayload.
+	
+	model receive: selectionMessage serialize.
+	
+	self assert: ( session collaborators at: remoteId ) currentSelectionAsDictionary =  (OrderedDictionary newFrom: {
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/testChangeSelectionUpdatesSelectionInCollaboratorsList.st:9]:
```suggestion
	self assert: ( session collaborators at: remoteId ) currentSelectionAsDictionary equals: (OrderedDictionary newFrom: {
```

File: packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/testUpdateSelectionFillsCCESelectionWithCorrectSelectionSymbols.st [hunk 58]
```diff
@@ -0,0 +1,18 @@
...
+		#selection -> ( OrderedDictionary newFrom: {
+			#categorySelection -> cceSelection categorySelection .
+			#classSelection -> cceSelection classSelection .
+			#methodSelection -> cceSelection methodSelection}) } ).
+	
+	
+	self assert: (  (networkInterface singleMessage payload at: #selection) = (expectedMessage payload at: #selection) ).
+	
+	networkInterface clear.
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/testUpdateSelectionFillsCCESelectionWithCorrectSelectionSymbols.st]: Please remove the empty lines.

File: packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/testUpdateSelectionTriggersChangeSelectionMessage.st [hunk 59]
```diff
@@ -0,0 +1,14 @@
...
+	expectedMessage := CCEMessage changeSelectionWith: ( OrderedDictionary newFrom: {
+		#id -> imageAccount humanReadableName .
+		#selection -> cceSelection } ).
+	
+	cceSelection updateSelectionWith: browser.
+	
+	self assert: ( networkInterface hasMessageOfType: ( CCEMessage clientMsgTypes at: #changeSelection:from: ) ).
+	
+	networkInterface clear.
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolSelectionCSTest.class/instance/testUpdateSelectionTriggersChangeSelectionMessage.st]: Remove the empty lines.

File: packages/CollabCodeEdit-Tests.package/CCEProtocolSessionCreationCSTest.class/instance/subClassSpecificTearDown.st [hunk 60]
```diff
@@ -0,0 +1,3 @@
+running
+subClassSpecificTearDown
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolSessionCreationCSTest.class/instance/subClassSpecificTearDown.st]: Remove the empty lines.

File: packages/CollabCodeEdit-Tests.package/CCEProtocolSessionCreationSSTest.class/instance/subClassSpecificTearDown.st [hunk 61]
```diff
@@ -0,0 +1,4 @@
+running
+subClassSpecificTearDown
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolSessionCreationSSTest.class/instance/subClassSpecificTearDown.st:3]: Remove the empty lines.

File: packages/CollabCodeEdit-Tests.package/CCEProtocolSessionJoinCSTest.class/instance/subClassSpecificTearDown.st [hunk 62]
```diff
@@ -0,0 +1,3 @@
+running
+subClassSpecificTearDown
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolSessionJoinCSTest.class/instance/subClassSpecificTearDown.st]: Remove the empty lines.

File: packages/CollabCodeEdit-Tests.package/CCEProtocolSessionJoinSSTest.class/instance/subClassSpecificTearDown.st [hunk 63]
```diff
@@ -0,0 +1,4 @@
+running
+subClassSpecificTearDown
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolSessionJoinSSTest.class/instance/subClassSpecificTearDown.st:3]: Remove the empty lines.

File: packages/CollabCodeEdit-Tests.package/CCEProtocolSessionLeaveSSTest.class/instance/subClassSpecificTearDown.st [hunk 64]
```diff
@@ -0,0 +1,4 @@
+running
+subClassSpecificTearDown
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolSessionLeaveSSTest.class/instance/subClassSpecificTearDown.st:3]: Remove the empty lines.

File: packages/CollabCodeEdit-Tests.package/CCEProtocolTestServerside.class/instance/subClassSpecificTearDown.st [hunk 65]
```diff
@@ -0,0 +1,3 @@
+running
+subClassSpecificTearDown
+	^ SubclassResponsibility
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolTestServerside.class/instance/subClassSpecificTearDown.st:3]:
```suggestion
	^ self subclassResponsibility
```

File: packages/CollabCodeEdit-Tests.package/CCEProtocolTestServerside.class/instance/tearDown.st [hunk 66]
```diff
@@ -2,5 +2,7 @@ running
 tearDown
 	networkInterface1 clear.
 	networkInterface2 clear.
-	networkInterface3 clear
+	networkInterface3 clear.
+	self subClassSpecificTearDown
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEProtocolTestServerside.class/instance/tearDown.st]: Remove the empty lines.

File: packages/CollabCodeEdit-Tests.package/CCESessionTest.class/instance/setUp.st [hunk 67]
```diff
@@ -5,4 +5,6 @@ setUp
 	account := CCEAccount new humanReadableName: testIdentifier.	
 	session imageAccount: account.
 	
-	testCollaborator := CCECollaborator new identifier: testIdentifier
\ No newline at end of file
+	testCollaborator := CCECollaborator new identifier: testIdentifier
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCESessionTest.class/instance/setUp.st]: Remove the empty lines.

File: packages/CollabCodeEdit-Tests.package/CCESessionTest.class/instance/testSetSelectionOfCollaborator.st [hunk 68]
```diff
@@ -0,0 +1,17 @@
...
+		#methodSelection -> #testMethod}. 
+		
+	self assert: testCollaborator currentSelection = nil.
+	
+	session addCollaboratorFromAccount: account.
+	
+	session setSelectionOfCollaborator: testIdentifier to: newSelection.
+	
+	self assert: (session collaborators at: testIdentifier) currentSelectionAsDictionary = newSelection.
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCESessionTest.class/instance/testSetSelectionOfCollaborator.st]: Remove the empty lines.

File: packages/CollabCodeEdit-Tests.package/CCESessionTest.class/instance/testSetSelectionOfCollaborator.st [hunk 69]
```diff
@@ -0,0 +1,17 @@
+tests collaborator managment
+testSetSelectionOfCollaborator	
+	| newSelection | 
+	
+	newSelection := OrderedDictionary newFrom: {
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCESessionTest.class/instance/testSetSelectionOfCollaborator.st:5]: Maybe that could be in the `setUp`.

File: packages/CollabCodeEdit-Tests.package/CCESessionTest.class/instance/testSetSelectionOfCollaborator.st [hunk 70]
```diff
@@ -0,0 +1,17 @@
...
+		#categorySelection -> #testCategory .
+		#classSelection -> #testClass .
+		#methodSelection -> #testMethod}. 
+		
+	self assert: testCollaborator currentSelection = nil.
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCESessionTest.class/instance/testSetSelectionOfCollaborator.st]: Please do not overuse newlines.

File: packages/CollabCodeEdit-Tests.package/CCESessionTest.class/instance/testSetSelectionOfCollaborator.st [hunk 71]
```diff
@@ -0,0 +1,17 @@
...
+		#classSelection -> #testClass .
+		#methodSelection -> #testMethod}. 
+		
+	self assert: testCollaborator currentSelection = nil.
+	
+	session addCollaboratorFromAccount: account.
+	
+	session setSelectionOfCollaborator: testIdentifier to: newSelection.
+	
+	self assert: (session collaborators at: testIdentifier) currentSelectionAsDictionary = newSelection.
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCESessionTest.class/instance/testSetSelectionOfCollaborator.st]:
```suggestion
	self assert: (session collaborators at: testIdentifier) currentSelectionAsDictionary equals: newSelection.
```

**Review by arminWel — DISMISSED** [review]

File: packages/CollabCodeEdit-Client.package/CCEBrowser.class/instance/editSelection..st (hunk 2 above)
Comment (simonstadlinger) [reply packages/CollabCodeEdit-Client.package/CCEBrowser.class/instance/editSelection..st]: in a setter method?;)

Comment (arminWel) [reply packages/CollabCodeEdit-Client.package/CCEBrowser.class/instance/editSelection..st]: Why not? :)

File: packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/instance/category..st (hunk 46 above)
Comment (m0e33) [reply packages/CollabCodeEdit-Core.package/CCEMethodRecategorizedEdit.class/instance/category..st]: probably, seems to be fine in my image.

**Pull request merged**